units/src/amount/error.rs

```diff
@@ -407,27 +423,26 @@ impl From<Infallible> for AmountDecoderError {
     fn from(never: Infallible) -> Self { match never {} }
 }
 
-#[cfg(feature = "encoding")]
-impl From<encoding::UnexpectedEofError> for AmountDecoderError {
```

Comment:
New pattern for me here so might be missing something, but would this From impl still be useful if updates to the new constructor?

Reply:
The reason for removing this is that if we keep it we have to commit to it, so we cannot stop using the `UnexpectedEofError`. As much as this is unlikely I'm trying to be as conservative as possible and since the from impl is only used internally there is no need for it other than ergonomics (in internal code). Also downstream should never be constructing `AmountDecoderError`.

Reply:
Got it, thanks for the explanation. Wasn't obvious to me how the `From` was part of the exported API.